forage-android/src/main/java/com/joinforage/forage/android/ecom/services/vault/bt/BTResponseParser.kt

```diff
@@ -15,48 +18,61 @@ internal class BTResponseParser(btRes: Result<Any?>) : VaultResponseParser {
-            // of *when* erroroneous BT response happen. Unfortunately,
-            // we do not currently track the specifics of the proxy_error
-            // that that BT returned to us.
-            // TODO: log the specific error that BT responds with when
-            //  resRegExp.containsProxyError == true
-            if (resRegExp.containsProxyError) UnknownErrorApiResponse else null
-        } catch (_: Exception) {
-            null
-        }
+        // In AbstractVaultSubmitter we track the forageError
+        // and the forage status code of the UnknownErrorApiResponse
+        // via the VaultProxyResponseMonitor. This keeps us informed
+        // of *when* erroroneous BT response happen. Unfortunately,
+        // we do not currently track the specifics of the proxy_error
+        // that that BT returned to us.
+        // TODO: log the specific error that BT responds with when
+        //  resRegExp.containsProxyError == true
+        return if (failureResponseRegExp.containsProxyError) UnknownErrorApiResponse else null
     }
 
-    private fun parseForageError(resRegExp: BaseResponseRegExp): ForageApiResponse.Failure? {
-        if (resRegExp.bodyText == null || resRegExp.statusCode == null) return null
-        return try {
-            ForageApiResponse.Failure(resRegExp.statusCode, resRegExp.bodyText)
-        } catch (_: Exception) {
+    private fun parseForageError(resRegExp: BtFailureResponseRegExp): ForageApiResponse.Failure? =
+        if (resRegExp.bodyText == null ||
+            resRegExp.statusCode == null ||
+            // if there's a proxy_error, don't try to parse as a Forage error
+            resRegExp.containsProxyError
+        ) {
             null
+        } else {
+            // if there's a body, a status code, and no proxy_error, then
+            // we embrace that the following line will throw an exception
+            ForageApiResponse.Failure(resRegExp.statusCode, resRegExp.bodyText)
         }
-    }
 
     private fun parseSuccessfulResponse(vaultResponse: Result<Any?>): ForageApiResponse.Success<String>? {
         return if (!isSuccessful) {
             null
         } else {
-            // note: Result.toString() wraps the actual response as
-            // "Success(<actual-value-here>)"
-            ForageApiResponse.Success(vaultResponse.getOrNull().toString())
+            // Basis Theory appears to leak their Gson dependency in
+            // the response as the value of Any? resolves to a
+            // com.google.gson.internal.LinkedTreeMap instance.
+            // Fortunately, this is of type Map<Any, Any?> so we can
+            // still parse it with JSONObject and avoid having to
+            // directory depend on Gson ourselves
+            val result = vaultResponse.getOrThrow()
+            val jsonResponse = when {
+                (result == null) -> JSONObject()
+                (result == "") -> JSONObject()
```

Comment:
These are to gracefully handle the deferred pin-submission flows since they technically have an empty body and were throwing with the new `API-VERSION: 2024-01-08`